### PR TITLE
[SPIR-V] Prefer llvm-spirv-<LLVM_VERSION_MAJOR> tool

### PR DESCRIPTION
**Pull request by linehill**
Prefer using `llvm-spirv-<LLVM_VERSION_MAJOR>` tool (i.e. `llvm-spirv-18`) over plain `llvm-spirv`. If the versioned tool is not found in PATH, fall back to use the plain `llvm-spirv`.

An issue with the using `llvm-spirv` is that the one found in PATH might be compiled against older LLVM version which could lead to crashes or obscure bugs. For example, `llvm-spirv` distributed by Ubuntu links against different LLVM version depending on the Ubuntu release (LLVM-10 in 20.04LTS, LLVM-13 in 22.04LTS).

### CODE REVIEW

**Comment by linehill** [comment]
@AnastasiaStulova, @MaskRay, gentle ping.

**Review by svenvh — APPROVED** [review]

**Comment by linehill** [comment]
Thanks for the review. Could you merge this PR on my behalf (I don't have write access)?

**Comment by svenvh** [comment]
> Thanks for the review. Could you merge this PR on my behalf (I don't have write access)?

Yes I am happy to do so, but would you mind rebasing this PR onto latest `main` first?  Just so that the checks can run again to avoid any potential new regressions since the checks ran last.

**Comment by linehill** [comment]
Ping, @svenvh. The patch has been rebased, good for landing?

**Comment by svenvh** [comment]
> Ping, @svenvh. The patch has been rebased, good for landing?

Thanks for the ping, it seems github didn't send me a notification when you rebased.

Just waiting for the final checks to complete, then I'll merge it.

**Comment by linehill** [comment]
Thanks, @svenvh.

**Pull request merged**
